crates/shared/src/price_estimation/competition.rs

```diff
@@ -184,17 +194,23 @@ fn successes<R, E>(results: &[(EstimatorIndex, Result<R, E>)]) -> usize {
 
 impl PriceEstimating for RacingCompetitionEstimator<Arc<dyn PriceEstimating>> {
     fn estimate(&self, query: Arc<Query>) -> futures::future::BoxFuture<'_, PriceEstimateResult> {
+        let out_token = match query.kind {
+            OrderKind::Buy => query.sell_token,
+            OrderKind::Sell => query.buy_token,
+        };
+        let context = self.ranking.provide_context(out_token);
```

Comment:
nanonit: maybe call this slightly different than the awaited context variable in the comparison closure to avoid confusion?